### PR TITLE
Adding support for region in credentials file

### PR DESCRIPTION
**Pull request by rayvafa**

### CODE REVIEW

**Comment by Nicktho** [comment]
it would be cool to generically support any additional settings found under each profile, since we dont really need to know the region.

File: lib/parseCredentials.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
 const parseProfileName = profileName => profileName.match(/\[(.*)\]/)[1];
-const getValue = line => line.match(new RegExp('= (.*)'))[1];
+const parseAttribute = line => {
+  const attributeKeyValue = line.replace(/ /g,'').split('=');
+  return {[attributeKeyValue[0]]: attributeKeyValue[1]};
+};
 
-module.exports = file => file
+module.exports = file => Object.assign({}, ...file
```
Comment (Nicktho) [inline lib/parseCredentials.js]: wizardry

Comment (rayvafa) [reply lib/parseCredentials.js]: yep, it's all magic ;)

**Comment by rayvafa** [comment]
@Nicktho have another look please

File: lib/parseCredentials.js [hunk 2]
```diff
@@ -1,13 +1,14 @@
 const parseProfileName = profileName => profileName.match(/\[(.*)\]/)[1];
-const getValue = line => line.match(new RegExp('= (.*)'))[1];
+const parseAttribute = line => {
+  const attributeKeyValue = line.replace(/ /g,'').split('=');
```
Comment (Nicktho) [inline lib/parseCredentials.js]:
```js
const [ attributeKey, attributeValue ] = line.replace(/ /g,'').split('=');
return { attributeKey: attributeValue };
```

might look a bit nicer here.

File: lib/parseCredentials.js [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
+  return {[attributeKeyValue[0]]: attributeKeyValue[1]};
+};
 
-module.exports = file => file
+module.exports = file => Object.assign({}, ...file
   .split('\n\n')
   .map(block => block.split('\n'))
```
Comment (Nicktho) [inline lib/parseCredentials.js]: does this still work with the change at https://github.com/Nicktho/awsprofiles/pull/2/files#diff-6ce893fb5c9c870be4c606ac6e7aa6d2R9

File: lib/parseCredentials.js [hunk 4]
```diff
@@ -1,13 +1,14 @@
 const parseProfileName = profileName => profileName.match(/\[(.*)\]/)[1];
-const getValue = line => line.match(new RegExp('= (.*)'))[1];
+const parseAttribute = line => {
+  const attributeKeyValue = line.replace(/ /g,'').split('=');
+  return {[attributeKeyValue[0]]: attributeKeyValue[1]};
+};
 
-module.exports = file => file
+module.exports = file => Object.assign({}, ...file
   .split('\n\n')
```
Comment (Nicktho) [inline lib/parseCredentials.js:8]: does this still work with the change at https://github.com/Nicktho/awsprofiles/pull/2/files#diff-6ce893fb5c9c870be4c606ac6e7aa6d2R9

File: lib/generateCredentials.js [hunk 5]
```diff
@@ -1,14 +1,9 @@
-const generateCredential = (name, awsAccessKeyId, awsSecretAccessKey) => `[${name}]
-aws_access_key_id = ${awsAccessKeyId}
-aws_secret_access_key = ${awsSecretAccessKey}`;
+const generateCredential = (name, profileCredentials) => `[${name}]
+${Object.keys(profileCredentials).map(credkey => `${credkey} = ${profileCredentials[credkey]}\n`).join('')}`;
```
Comment (Nicktho) [inline lib/generateCredentials.js]: nested `${}` might be a bit hard to follow, and since all were using the string template for is a new line, perhaps something like

```js
const generateCredential = (name, profileCredentials) => [`[${name}]`]
  .concat(
    Object.keys(profileCredentials).map(credkey => `${credkey} = ${profileCredentials[credkey]}`)
  ).join('\n') \\ Each line in paragraph
```

File: lib/generateCredentials.js [hunk 6]
```diff
@@ -1,14 +1,9 @@
...
-      ...lines,
-      generateCredential(profile, awsAccessKeyId, awsSecretAccessKey),
-    ];
-  }, [])
-  .join('\n\n');
+  .reduce((lines, profile) => [
+    ...lines,
+    generateCredential(profile, credentials[profile]),
+  ], [])
+  .join('\n');
```
Comment (Nicktho) [inline lib/generateCredentials.js]: this will need to be `\n\n` with the prev change

**Comment by rayvafa** [comment]
@Nicktho sorry for the big delay in finishing this off. please have a look again and let me know if there is any issues. cheers

**Comment by Nicktho** [comment]
Nice! 🎉
